pkg/spdk/replica.go

```diff
@@ -218,7 +229,40 @@ func (r *Replica) ValidateAndUpdate(bdevLvolMap map[string]*spdktypes.BdevInfo)
 		}
 	}
 
-	// TODO: Need to validate IP and Port if the replica is exposed
+	if r.IP == "" {
+		return fmt.Errorf("found invalid IP %s for replica %s", r.IP, r.Name)
+	}
+
+	if r.IsExposed {
+		return nil
+	}
+
+	if r.PortStart == 0 || r.PortEnd == 0 {
+		return fmt.Errorf("found invalid Ports [%d, %d] for the exposed replica %s", r.PortStart, r.PortEnd, r.Name)
+	}
+
+	nqn := helpertypes.GetNQN(r.Name)
+	subsystem := subsystemMap[nqn]
+	if subsystem == nil || len(subsystem.ListenAddresses) == 0 {
+		return fmt.Errorf("cannot find the Nvmf subsystem with NQN %s for the exposed replica %s", nqn, r.Name)
+	}
+	exposedPort := 0
+	for _, listenAddr := range subsystem.ListenAddresses {
+		if listenAddr.Adrfam != spdktypes.NvmeAddressFamilyIPv4 || listenAddr.Trtype != spdktypes.NvmeTransportTypeTCP {
+			continue
+		}
+		exposedPort, err = strconv.Atoi(listenAddr.Trsvcid)
+		if err != nil {
+			return err
+		}
+		break
+	}
+	if int32(exposedPort) != r.PortStart {
+		if stopErr := spdkClient.StopExposeBdev(nqn); stopErr != nil {
```

Comment:
Do we need a lock protection for a spdkClient operation?

Reply:
I was thinking the same question before, but I am fine with it fine for now.

Reply:
Sure. I will add it after this PR merge